core-common/src/main/java/org/mapstruct/CollectionMappingStrategy.java

```diff
@@ -51,5 +52,16 @@
      * Identical to {@link #SETTER_PREFERRED}, only that adder methods will be preferred over setter methods, if both
      * are present for a given collection-typed property.
      */
-    ADDER_PREFERRED;
+    ADDER_PREFERRED,
+
+    /**
+     * Identical to {@link #SETTER_PREFERRED}, however the target collection will not be cleared and accessed via
+     * addAll.
+     * <pre>
+     * orderDto.getOrderLines().clear();
+     * orderDto.getOrderLines().addAll(order.getOrderLines);
+     * </pre>.
+     * Instead the target accessor (e.g. set) will be used on the target bean to set the collection.
+     */
+    TARGET_IMMUTABLE;
```

Comment:
Good point. Thought about that one. I'd like to express that the use case related to immutable collections.  I'd like to stick to TARGET_IMMUTABLE (if you don't have any objections).